certbot_dns_loopia/_internal/dns_loopia.py

```diff
@@ -0,0 +1,270 @@
+"""
+Contains the Loopia DNS ACME authenticator class.
+"""
+import logging
+import xmlrpc.client
+from dataclasses import dataclass
+from functools import wraps
+from typing import Callable, Optional, Tuple, get_type_hints, Any, TypeVar, cast, Mapping, List
+from xmlrpc.client import ServerProxy
+
+from certbot.configuration import NamespaceConfig
+from certbot.errors import PluginError
+from certbot.plugins.dns_common import DNSAuthenticator, CredentialsConfiguration
+from tldextract import TLDExtract
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, eq=False)
+class DnsRecord:
+    """
+    Matches the "record_obj" as defined in the Loopia API spec.
+    See https://www.loopia.se/api/record_obj/
+
+    The names of the properties exactly match the API spec, to
+    enable seamless usage with the API.
+    """
+    type: str
+    ttl: int = 3600
+    priority: int = 0
+    rdata: str = ""
+    record_id: int = 0
+
+    def __post_init__(self) -> None:
+        type_hints = get_type_hints(self)
+
+        for name, type_hint in type_hints.items():
+            value = getattr(self, name)
+
+            if not isinstance(value, type_hint):
+                raise ValueError(
+                    f"Expected attribute '{name}' to be of type "
+                    f"{type_hint}, but value was {value}"
+                )
+
+    def __eq__(self, other: object) -> bool:
+        """
+        Special equals operator implementation that does not compare record ID.
+        """
+        assert isinstance(other, DnsRecord)
```

Comment:
Maybe we should not assert here but rather return `False` if `other` is not an instance of `DnsRecord`.